packages/api/src/external/hie/register-facility.ts.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  AddressStrict,
+  AddressWithCoordinates,
+  removeCoordinates,
+} from "@metriport/core/domain/location-address";
+import {
+  FacilityRegister,
+  Facility,
+  FacilityCreate,
+  isOboFacility,
+} from "../../domain/medical/facility";
+import { buildCqOrgName } from "../../external/carequality/shared";
+import { buildCwOrgName } from "../../external/commonwell/shared";
+import { getCxOrganizationNameOidAndType } from "../../command/medical/organization/get-organization";
+import { getAddressWithCoordinates } from "../../domain/medical/address";
+import { getCqOboData } from "../../external/carequality/get-obo-data";
+import { createOrUpdateFacility } from "../../command/medical/facility/create-or-update-facility";
+import { createOrUpdateInCq, createOrUpdateInCw } from "./shared";
+
+/**
+ * Registers a new facility within HIEs.
+ *
+ * @param cxId
+ * @param facility
+ * @returns The updated facility.
+ */
+export async function registerFacilityWithinHIEs(
+  cxId: string,
+  facility: FacilityRegister
+): Promise<Facility> {
+  const isObo = isOboFacility(facility.type);
```

Comment:
I'm thinking we should probably also check if the Org is `healthcare_it_vendor`, right?

Reply:
My assumption was that this endpoint was always being triggered for `healthcare_it_vendor` but to add an extra layer of checks i can add this